src/components/Form/Form.tsx

```diff
@@ -35,38 +37,67 @@ const Form: React.FC<FormProps> = ({
         }
     };
 
-    const optionAdded = (index: number, opt: string, isRadio: boolean) => {
-        if (!data[index].selected || isRadio) {
-            data[index].selected = [opt];
+    const optionAdded = (
+        elementIndex: number,
+        opt: string | number,
+        isRadio: boolean,
+    ) => {
+        if (isRadio) {
+            // @ts-ignore:next-line
+            const selectRadioValue = data[elementIndex].options[Number(opt)];
+            if (typeof selectRadioValue === 'string') {
+                data[elementIndex].selected = [selectRadioValue];
+            } else {
+                data[elementIndex].selectedCard = selectRadioValue;
+            }
+            return;
+        }
+
+        if (!data[elementIndex].selected) {
+            data[elementIndex].selected = [String(opt)];
         } else {
-            data[index]?.selected?.push(opt);
+            data[elementIndex]?.selected?.push(String(opt));
         }
     };
 
     const optionToggled = (
         event: React.ChangeEvent<HTMLInputElement>,
-        index: number,
+        elementIndex: number,
         opt: string,
     ) => {
         const isRadio = Boolean(event.target.type === 'radio');
 
         if (event.target.checked) {
-            optionAdded(index, opt, isRadio);
+            optionAdded(elementIndex, opt, isRadio);
         } else {
-            const filtered = data[index]?.selected?.filter(
+            const filtered = data[elementIndex]?.selected?.filter(
                 (item) => item !== opt,
             );
-            data[index].selected = filtered;
+            data[elementIndex].selected = filtered;
```

Comment:
can simply it by:
`data[elementIndex].selected = data[elementIndex]?.selected?.filter(
      (item) => item !== opt,
 );`

Reply:
Do you think having a one liner is more simple?
I agree its less code, but I added the extra lines to make it more simple for the next dev to understand.
What do you think?